app/src/main/java/com/android/halanvendordiscovery/domain/details/useCase/GetMerchantsUseCase.kt

```diff
@@ -0,0 +1,11 @@
+package com.android.halanvendordiscovery.domain.details.useCase
+
+import com.android.halanvendordiscovery.domain.details.model.MerchantModel
+import com.android.halanvendordiscovery.domain.details.repo.DetailsRepository
+import javax.inject.Inject
+
+class GetMerchantsUseCase @Inject constructor(private val repository: DetailsRepository) {
```

Comment:
this `private val repository: DetailsRepository` , having the reference named repository because the type describes what type it is ,is good 👍🏾